tests/firedrake/regression/test_solving_interface.py

```diff
@@ -262,6 +257,35 @@ def test_solve_empty_form_rhs(mesh):
     assert errornorm(x, w) < 1E-10
 
 
+def test_solve_assembled_lhs(mesh):
```

Comment:
Check both explicit and matrix free matrices.
```suggestion
@pytest.mark.parametrize("mat_type", ["aij", "matfree"])
def test_solve_assembled_lhs(mesh, mat_type):
```